include/graphblas/algorithms/amg/amg.hpp

```diff
@@ -0,0 +1,230 @@
+
+/*
+ *   Copyright 2021 Huawei Technologies Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file amg.hpp
+ * @author Alberto Scolari (alberto.scolari@huawei.com)
+ * @brief File with the main routine to run a full AMG simulation, comprising multi-grid runs
+ *        with Red-Black Gauss-Seidel smoothing.
+ * @date 2021-04-30
+ */
+
+#ifndef _H_GRB_ALGORITHMS_AMG
+#define _H_GRB_ALGORITHMS_AMG
+
+#include <graphblas.hpp>
+
+#include "amg_data.hpp"
+#include "multigrid_v_cycle.hpp"
+
+#include <utils/print_vec_mat.hpp>
+
+namespace grb {
+	namespace algorithms {
+
+
+
+		
+		/**
+		 * @brief High-Performance Conjugate Gradient algorithm implementation running entirely on GraphBLAS.
```

Comment:
Remove  `@brief` and check that the line length doesn't exceed approx. 80 chars (excluding leading tabs and spaces)